python/cudf_polars/tests/utils/test_broadcast.py

```diff
@@ -0,0 +1,74 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import pytest
+
+import cudf._lib.pylibcudf as plc
+
+from cudf_polars.containers import NamedColumn
+from cudf_polars.dsl.ir import broadcast
+
+
+@pytest.mark.parametrize("target", [4, None])
+def test_broadcast_all_scalar(target):
+    columns = [
+        NamedColumn(
+            plc.column_factories.make_numeric_column(
+                plc.DataType(plc.TypeId.INT8), 1, plc.MaskState.ALL_VALID
+            ),
+            f"col{i}",
+        )
+        for i in range(3)
+    ]
+    result = broadcast(*columns, target_length=target)
+    expected = 1 if target is None else target
+
+    assert all(column.obj.size() == expected for column in result)
+
+
+def test_invalid_target_length():
+    columns = [
+        NamedColumn(
+            plc.column_factories.make_numeric_column(
+                plc.DataType(plc.TypeId.INT8), 4, plc.MaskState.ALL_VALID
+            ),
+            f"col{i}",
+        )
+        for i in range(3)
+    ]
+    with pytest.raises(RuntimeError):
+        _ = broadcast(*columns, target_length=8)
```

Comment:
Any reason to prefer this over implicitly discarding?
```suggestion
        broadcast(*columns, target_length=8)
```

Reply:
a ruff lint, I think, don't recall which one.